.github/workflows/ci.yml

```diff
@@ -22,33 +20,99 @@ jobs:
       # and hyphens.
       run: echo "TAG=branch-${GITHUB_REF_NAME//[^A-Za-z0-9._-]/-}" | tee -a $GITHUB_ENV
 
-    - run: ./devel/start-localhost-registry $REGISTRY_PORT
+    # GITHUB_TOKEN is unreliable¹ so use a token from nextstrain-bot.
+    # ¹ https://github.com/docker/build-push-action/issues/463#issuecomment-939394233
+    - uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: nextstrain-bot
+        password: ${{ secrets.GH_TOKEN_NEXTSTRAIN_BOT_MANAGE_PACKAGES }}
 
-    - run: ./devel/build -r localhost:$REGISTRY_PORT -t $TAG
+    - run: ./devel/build -r ghcr.io -t $TAG
 
-    - if: github.event_name != 'pull_request'
-      uses: docker/login-action@v1
-      with:
-        username: ${{ secrets.DOCKER_LOGIN }}
-        password: ${{ secrets.DOCKER_PASSWORD }}
+    outputs:
+      tag: ${{ env.TAG }}
+
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
 
-    - if: github.event_name != 'pull_request' && startsWith(env.TAG, 'branch-')
-      name: Copy $TAG images to Docker Hub (non-default branch)
-      run: ./devel/copy-images -i localhost:$REGISTRY_PORT -o docker.io -t $TAG
+    - uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: nextstrain-bot
+        password: ${{ secrets.GH_TOKEN_NEXTSTRAIN_BOT_MANAGE_PACKAGES }}
```

Comment:
Login is needed at least because the intermediate packages are currently private. I'm not sure how it would behave if it were public.

We shouldn't need `delete:packages` or `write:packages` for pulling, but I didn't want to create another token with only `read:packages`. That was mostly out of laziness, so if you think a separate token is better then I can do that.

Reply:
Nope, I think that's fine.